src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2149,6 +2156,11 @@ void RTPSParticipantImpl::deleteAllUserEndpoints()
         }
 #endif // if HAVE_SECURITY
 
+        if (kind == READER)
+        {
+            static_cast<BaseReader*>(endpoint)->local_actions_on_reader_removed();
```

Comment:
Should not be `dynamic_cast`?

Reply:
The collections always store `BaseReader*` so it is well known to static_cast at this point